packages/babel-plugin-mui-css/tests/index.test.js

```diff
@@ -0,0 +1,312 @@
+const INPUT_VARS = resolve(RUNTIME_DIR, './vars.ts');
+const OUTPUT_DIR = resolve(RUNTIME_DIR, './output');
+
+// Prettier messes with the code strings.
+// prettier-ignore
+describe('babel-plugin-mui-css', () => {
+  beforeEach(() => {
+    rimraf.sync(RUNTIME_DIR);
+    mkdirp.sync(resolve(RUNTIME_DIR, 'src'));
+    mkdirp.sync(resolve(RUNTIME_DIR, 'output'));
+  });
+
+  after(() => {
+    rimraf.sync(RUNTIME_DIR);
+  });
+
+  it('works', () => {
+    transform({
+      files: [{
+        name: 'index.js',
+        input: js`
+          const styles = css('MuiDataGrid-panel', {
+            root: {
+              border: '1px solid black',
+            },
+            focused: {
+              border: '1px solid blue',
+            },
+          })
+        `,
+        output: js`
+          import "./index.autogenerated.css";
+          const styles = {
+            "root": "MuiDataGrid-panel",
+            "focused": "MuiDataGrid-panel--focused"
+          };
+        `,
+        css: css`
+          .MuiDataGrid-panel { border:1px solid black; }
+          .MuiDataGrid-panel--focused { border:1px solid blue; }
+        `
+      }],
+    });
+  });
+
+  it('transforms nested selectors', () => {
+    transform({
+      files: [{
+        name: 'index.js',
+        input: js`
+          const styles = css('MuiDataGrid-panel', {
+            root: {
+              '#id': {
+                color: 'black',
+              },
+              '.class-a': {
+                color: 'black',
+              },
+              '&.class-b': {
+                color: 'black',
+              },
+              '> input': {
+                color: 'black',
+              },
+            }
+          })
+        `,
+        output: js`
+          import "./index.autogenerated.css";
+          const styles = {
+            "root": "MuiDataGrid-panel"
+          };
+        `,
+        css: css`
+          .MuiDataGrid-panel {  }
+          .MuiDataGrid-panel #id { color:black; }
+          .MuiDataGrid-panel .class-a { color:black; }
+          .MuiDataGrid-panel.class-b { color:black; }
+          .MuiDataGrid-panel > input { color:black; }
+        `,
+      }]
+    });
+  });
+
+  it('supports CSS variables', () => {
+    transform({
+      vars: ts`
+        export const vars = {
+          color: {
+            primary: 'var(--color-primary)',
+          },
+        } as const;
+      `,
+      files: [{
+        name: 'index.js',
+        input: js`
+          const styles = css('MuiDataGrid-panel', {
+            root: {
+              color: vars.color.primary,
+            },
+          })
+        `,
+        output: js`
+          import "./index.autogenerated.css";
+          const styles = {
+            "root": "MuiDataGrid-panel"
+          };
+        `,
+        css: css`
+          .MuiDataGrid-panel { color:var(--color-primary); }
+        `,
+      }]
+    });
+  });
+
+  it('supports minification', () => {
```

Comment:
I'm not sure if this is a use-case we want to support, distributing the CSS un-minified (like we do for JS) also makes sense. I could remove this feature.